### PR TITLE
Decode Unsigned Tx with Codec

### PR DESCRIPTION
**Pull request by h0ngcha0**
Use `codec` to decode unsigned transaction when `buildUnsignedTx`

### CODE REVIEW

File: packages/web3/src/signer/tx-builder.ts [hunk 1]
```diff
@@ -113,16 +115,16 @@ export abstract class TransactionBuilder {
...
     return {
-      fromGroup: decoded.fromGroup,
-      toGroup: decoded.toGroup,
+      fromGroup: fromGroup,
+      toGroup: toGroup,
       unsignedTx: params.unsignedTx,
-      txId: decoded.unsignedTx.txId,
-      gasAmount: decoded.unsignedTx.gasAmount,
-      gasPrice: fromApiNumber256(decoded.unsignedTx.gasPrice)
+      txId: UnsignedTxCodec.txId(decoded),
```
Comment (Lbqds) [inline packages/web3/src/signer/tx-builder.ts:125]: The `UnsignedTx.txId` will encode the tx. Maybe we can directly calculate the tx id using the raw tx bytes.

File: packages/web3/src/transaction/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import { web3 } from '../'
+import { PrivateKeyWallet } from '@alephium/web3-wallet'
+
+describe('transaction utils', () => {
+  beforeAll(() => {
+    web3.setCurrentNodeProvider('http://127.0.0.1:22973', undefined, fetch)
+  })
+
+  it('should calculate the group of transfer transaction', async () => {
+    const [signer1, signer2] = await getSigners(2)
```
Comment (Lbqds) [inline packages/web3/src/transaction/utils.test.ts:35]: The `getSigners` function returns group 0 accounts by default. Maybe we can test the case where `fromGroup` and `toGroup` are not the same.

**Review by Lbqds — APPROVED** [review]
LGTM 👍

File: packages/web3/src/utils/group.ts
Comment (polarker) [inline packages/web3/src/utils/group.ts:1]: This seems better to be in `address` module?

File: packages/web3/src/utils/group.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { TOTAL_NUMBER_OF_GROUPS } from '../constants'
+import djb2 from './djb2'
+
+export function groupFromBytes(bytes: Uint8Array): number {
```
Comment (polarker) [inline packages/web3/src/utils/group.ts:22]: I think this one should be a private function in `address`, as it's only used for asset addresses. The hints for contracts are different.

We could also rename it to be more informative: `groupFromBytesForAssetAddress()`

File: packages/web3/src/signer/tx-builder.ts (hunk 1 above)
Comment (h0ngcha0) [reply packages/web3/src/signer/tx-builder.ts]: Fixed, thanks :+1:

File: packages/web3/src/utils/group.ts (hunk 3 above)
Comment (h0ngcha0) [reply packages/web3/src/utils/group.ts]: Fixed, thanks :+1:

**Pull request merged**